### PR TITLE
common section card and basic root section card with search header

### PR DESCRIPTION
**Pull request by danoswaltCL**
This is a basic implementation of the root section card with adjustments to the common section card for header-left and header-right. This is a very WIP commit so we can have something to view while we hook up the inner components. You'll see that the right side and content are placeholders until we sew in the action buttons element that @zackcl is working on.

This also sews in the search header, although it isn't really hooked up yet, just wanted to see it in there 👍 @Yagnik56 

image

I also put in the pieces for the paginated call action and the selectors we'll need to show spinner and load up the flags on load of this section card.

Please see the mock call in `featureFlagService.ts` that is in here temporarily to send mock flag data for us to use until we've merged in the endpoint changes, I don't have the mock-server PR ready because it's not cooperating, but perhaps this is plenty good enough.

```
fetchFeatureFlags(fromStarting?: boolean) {
    this.store$.dispatch(FeatureFlagsActions.actionFetchFeatureFlags({ fromStarting }));

    // mock response
    setTimeout(() => {
      this.store$.dispatch(
        FeatureFlagsActions.actionFetchFeatureFlagsSuccess({
          flags: mockFeatureFlags,
          totalFlags: 1,
        })
      );
    }, 3000);
  }
```

### CODE REVIEW

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-section-card/feature-flag-root-section-card.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<app-common-section-card>
+  <app-common-section-card-search-header 
+    header-left
+    [filterOptions]="['All', 'Name', 'Status', 'Context']"
+    [initialFilterOption]="'All'"
+    [initialSearchValue]=""
+    (search)="onSearch($event)"
+  ></app-common-section-card-search-header>
+  <div header-right>right</div>
```
Comment (Yagnik56) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-section-card/feature-flag-root-section-card.component.html:9]: both header components should be in one NG container, right?

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-section-card/feature-flag-root-section-card.component.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  <div content *ngIf="!(isLoadingFeatureFlags$ | async); else loading"><div content *ngIf="!(isLoadingFeatureFlags$ | async); else loading"><pre>{{( allFeatureFlags$ | async) | json }}</pre></div></div>
+  <ng-template #loading>
```
Comment (Yagnik56) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-section-card/feature-flag-root-section-card.component.html:11]: why `*ngIf="!(isLoadingFeatureFlags$ | async); else loading"` condition div is here twice?

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-page-content.component.html [hunk 3]
```diff
@@ -1,3 +1 @@
-<app-common-section-card-list>
-    <div section-card>Hi I'm a section card 1</div>
-</app-common-section-card-list>
+
```
Comment (Yagnik56) [inline frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-page-content.component.html:1]: are we going to use this component?

Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-page-content/feature-flag-root-page-content.component.html:1]: will be removed by other pr

File: frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-section-card/feature-flag-root-section-card.component.html (hunk 1 above)
Comment (danoswaltCL) [reply frontend/projects/upgrade/src/app/features/dashboard/feature-flags/pages/feature-flag-root-page/feature-flag-root-section-card/feature-flag-root-section-card.component.html:9]: like with the section-list, we chatted in slack and decided that an extra layer here wasn't necessary. That kind of change can be made pretty easily in the future to pull out code into it's own component later  if we find ourselves needing a thing like a second type of header that doesn't work with a left-right layout.

**Comment by danoswaltCL** [comment]
oops forgot to do this: https://github.com/CarnegieLearningWeb/UpGrade/pull/1552/commits/a81548aa4fe41d7defb0bed40be491ad60695e97 maybe that was making this confusing 🙃

**Review by zackcl — APPROVED** [review]

**Pull request merged**